codersdk/deployment.go

```diff
@@ -3420,6 +3423,17 @@ Write out the current server config as YAML to stdout.`,
 			YAML:        "api_keys",
 			Annotations: serpent.Annotations{}.Mark(annotationFormatDuration, "true"),
 		},
+		{
+			Name:        "Workspace Agent Logs Retention",
+			Description: "How long workspace agent logs are retained. Logs from non-latest workspace builds are deleted after this period to free up storage space. Set to 0 to disable automatic deletion of workspace agent logs.",
+			Flag:        "workspace-agent-logs-retention",
+			Env:         "CODER_WORKSPACE_AGENT_LOGS_RETENTION",
+			Value:       &c.Retention.WorkspaceAgentLogs,
+			Default:     "7d",
```

Comment:
What was your thought process for 7d? Is this table particularly egregious in its disk usage?

Reply:
That was a copy paste comment from existing code, already fixed [`fe71946` (#21039)](https://github.com/coder/coder/pull/21039/commits/fe71946cf182e2477954e0ef6498fc9c12d3624e). 😉

I think 7d is fine considering the constraint:
- Latest build logs are _always_ kept
- Agent connection is considered

If you want to debug a build, it's quite likely that the logs will be present or old enough that you don't care.

Unfortunately, the original PR did not add much motivation for the value: https://github.com/coder/coder/pull/6558, but given this has been the value since it's inception, I'm OK with it staying this way.